File: lib-base/src/main/java/com/llj/lib/base/ILoadingDialogHandler.java
package com.llj.lib.base;

import com.llj.lib.net.observer.ITag;

import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Action;
import io.reactivex.functions.Consumer;

/**
 * ArchitectureDemo
 * describe: 供activity和dialog使用，
 * author liulj
 * date 2018/5/24
 */
public interface ILoadingDialogHandler<T extends
        Disposable> extends Consumer<T>,
        Action,
        ITag {
    ITag initLoadingDialog();

    ITag getLoadingDialog();

    default void showLoadingDialog() {
    }

    default void dismissLoadingDialog() {
    }

    //显示
    @Override
    default void accept(Disposable o) throws Exception {
        showLoadingDialog();
    }

    //隐藏
    @Override
    default void run() throws Exception {
        dismissLoadingDialog();
    }
}

File: lib-base/src/main/java/com/llj/lib/base/ILoadingDialogHandler.kt
package com.llj.lib.base

import android.content.DialogInterface
import com.llj.lib.net.observer.ITaskId

import io.reactivex.disposables.Disposable
import io.reactivex.functions.Action
import io.reactivex.functions.Consumer

/**
 * ArchitectureDemo
 *
 * describe: 供activity控制dialog
 *
 * @author llj
 * @date 2018/5/24
 */
interface ILoadingDialogHandler<T> : Consumer<Disposable>, Action, ITaskId {

  fun getLoadingDialog(): T?

  fun initLoadingDialog(): T?

  fun showLoadingDialog()

  fun dismissLoadingDialog()

  //取消回调监听
  fun onLoadingDialogCancel(dialog: DialogInterface?)
  fun onLoadingDialogDismiss(dialog: DialogInterface?)

  //显示
  override fun accept(disposable: Disposable?) {
    showLoadingDialog()
  }

  //隐藏
  override fun run() {
    dismissLoadingDialog()
  }
}
